internal/security/proxy/service.go

```diff
@@ -319,6 +319,10 @@ func (s *Service) postCert(cp bootstrapConfig.CertKeyPair) *CertError {
 	}
 	tokens := []string{s.configuration.KongURL.GetProxyBaseURL(), CertificatesPath}
 	req, err := http.NewRequest(http.MethodPost, strings.Join(tokens, "/"), strings.NewReader(string(data)))
+	if err != nil {
+		s.loggingClient.Error(err.Error())
+		return &CertError{err.Error(), InternalError}
```

Comment:
Please add context to the error, like err = fmt.Errorf( "failed to upload Kong certificate... : %v", err)

Reply:
Yes, it is required because of golang convention.  The issue is not the lack of an error check.  The issue is that by convention, when you return an object and an error, when there is an error, you return nil as the object, so that you don't accidentially operate on an uninitialized object.  Thus, req.Header.Add could panic(), because in the original code, err is not checked until after req.Header is accessed.

Reply:
The error check on line 327 (after req.Header.Add) is not needed if the error is checked before req.header.add, because req.header.add never fails.